lib/srv/sshserver.go

```diff
@@ -878,7 +878,7 @@ func (s *Server) handleAgentForward(ch ssh.Channel, req *ssh.Request, ctx *ctx)
 	ctx.setAgent(clientAgent, authChan)
 
 	pid := os.Getpid()
-	socketPath := filepath.Join(os.TempDir(), fmt.Sprintf("teleport-agent-%v.socket", uuid.New()))
+	socketPath := filepath.Join(os.TempDir(), fmt.Sprintf("teleport-agent-%v.socket", uuid.New()[0:8]))
```

Comment:
@russjones let's go with option 2. because it protects us from collisions on the implementation level - AFAIK golang checks for collisions